### PR TITLE
boards: shields: Clean up Kconfig.defconfig files

### PR DESCRIPTION
**Pull request by erwango**
Shield's Kconfig.defconfig files should only be used to configure components options, or drivers specific (eg init priority, ..), depending on driver activation which remains application responsibility. High level components (such as DISPLAY) should not be enabled by default on boards activation (which is similar to policy used for boards).

Additionally, clean up busses activation (SPI, I2C), which is not required as bus activation is already done automatically on bus devices activation.

### CODE REVIEW

File: boards/shields/npm1100_ek/Kconfig.defconfig [hunk 1]
```diff
@@ -1,9 +0,0 @@
-# Copyright (c) 2023 Nordic Semiconductor ASA
-# SPDX-License-Identifier: Apache-2.0
-
-if SHIELD_NPM1100_EK
-
-config REGULATOR
-	default y
-
```
[the remark this answers is not in the document]
Comment (gmarull) [reply boards/shields/npm1100_ek/Kconfig.defconfig:8]: Indeed, you're totally right.

File: boards/shields/npm1100_ek/Kconfig.defconfig [hunk 2]
```diff
@@ -1,9 +0,0 @@
-# Copyright (c) 2023 Nordic Semiconductor ASA
-# SPDX-License-Identifier: Apache-2.0
-
-if SHIELD_NPM1100_EK
-
-config REGULATOR
```
Comment (gmarull) [inline boards/shields/npm1100_ek/Kconfig.defconfig:6]: if it's not too much to ask, could you add CONFIG_REGULATOR=y to https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/shields/npm6001_ek/prj.conf

**Review by gmarull — CHANGES_REQUESTED** [review]

File: boards/shields/npm1100_ek/Kconfig.defconfig (hunk 2 above)
Comment (erwango) [reply boards/shields/npm1100_ek/Kconfig.defconfig:6]: Sure, thanks for the heads up.

File: samples/shields/npm1300_ek/prj.conf [hunk 3]
```diff
@@ -3,3 +3,4 @@
 
 CONFIG_SHELL=y
 CONFIG_LOG=y
+CONFIG_REGULATOR=y
```
Comment (gmarull) [inline samples/shields/npm1300_ek/prj.conf:6]: missing gpio and sensor here

File: samples/shields/npm6001_ek/prj.conf [hunk 4]
```diff
@@ -5,3 +5,4 @@ CONFIG_SHELL=y
 CONFIG_SHELL_GETOPT=y
 CONFIG_GETOPT=y
 CONFIG_GETOPT_LONG=y
+CONFIG_REGULATOR=y
```
Comment (gmarull) [inline samples/shields/npm6001_ek/prj.conf:8]: `CONFIG_GPIO=y`, `CONFIG_WATCHDOG=y`

File: boards/shields/max3421e/Kconfig.defconfig [hunk 5]
```diff
@@ -1,9 +0,0 @@
-# Copyright (c) 2022 Nordic Semiconductor ASA
-# SPDX-License-Identifier: Apache-2.0
-
-if SHIELD_SPARKFUN_MAX3421E
-
-config UHC_DRIVER
```
Comment (erwango) [inline boards/shields/max3421e/Kconfig.defconfig:6]: @jfischer-no I'd like you to verify on this specific change.
To me, there is no reason this driver doesn't follow the same rule as others

Comment (jfischer-no) [reply boards/shields/max3421e/Kconfig.defconfig:6]: Change is correct, thanks.

**Review by jfischer-no — DISMISSED** [review]

File: samples/shields/npm1300_ek/prj.conf (hunk 3 above)
Comment (erwango) [reply samples/shields/npm1300_ek/prj.conf:6]: `CONFIG_SENSOR=y` added, but I omit GPIO, which is part of default boards configuration (so theoretically should always be there).
But I can add it, if you think there's a risk it is finally missing.

**Review by gmarull — DISMISSED** [review]

**Review by stephanosio — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**